Unable to build ClickHouse from source on macOS Monterey

> Make sure that `git diff` result is empty and you've just pulled fresh master. Try cleaning up cmake cache. Just in case, official build instructions are published here: https://clickhouse.com/docs/en/development/build/

- [x] git diff is empty
- [x] Pulled fresh master
- [x] Cleaned `cmake` cache
- [x] Re-reviewed build instructions for Max OSX multiple times    
- [x] git submodules re-init and up to date 

**Operating system**

> OS kind or distribution, specific version/release, non-standard kernel if any. If you are trying to build inside virtual machine, please mention it too.

Mac OS Monterey, version 12.3.1

**Cmake version**
3.23.1

**Ninja version**
1.10.2

**Compiler name and version**

```
#>> $(brew --prefix llvm)/bin/clang --version
Homebrew clang version 13.0.1
Target: x86_64-apple-darwin21.4.0
Thread model: posix
InstalledDir: /usr/local/opt/llvm/bin
```

```
#>> $(brew --prefix llvm)/bin/clang++ --version
Homebrew clang version 13.0.1
Target: x86_64-apple-darwin21.4.0
Thread model: posix
InstalledDir: /usr/local/opt/llvm/bin
```

**Full cmake and/or ninja output**

```
#>> cmake -DCMAKE_C_COMPILER=$(brew --prefix llvm)/bin/clang -DCMAKE_CXX_COMPILER=$(brew --prefix llvm)/bin/clang++ -DCMAKE_AR=$(brew --prefix llvm)/bin/llvm-ar -DCMAKE_RANLIB=$(brew --prefix llvm)/bin/llvm-ranlib -DOBJCOPY_PATH=$(brew --prefix llvm)/bin/llvm-objcopy -DCMAKE_BUILD_TYPE=RelWithDebInfo ..

-- The C compiler identification is Clang 13.0.1
-- The CXX compiler identification is Clang 13.0.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/local/opt/llvm/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/local/opt/llvm/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- The ASM compiler identification is Clang with GNU-like command-line
-- Found assembler: /usr/local/opt/llvm/bin/clang
Homebrew clang version 13.0.1
Target: x86_64-apple-darwin21.4.0
Thread model: posix
InstalledDir: /usr/local/opt/llvm/bin
-- Using objcopy: /usr/local/opt/llvm/bin/llvm-objcopy
-- Using strip: /usr/bin/strip
-- Found Git: /usr/bin/git (found version "2.30.1 (Apple Git-130)") 
-- HEAD's commit hash 7dc084419e16c52650088a3ae948850d94c4edf5
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
  (commit or discard the untracked or modified content in submodules)
        modified:   contrib/rapidjson (untracked content)

no changes added to commit (use "git add" and/or "git commit -a")
CMake Warning at cmake/ccache.cmake:17 (message):
  CCache is not found.  We recommend setting it up if you build ClickHouse
  from source often.  Setting it up will significantly reduce compilation
  time for 2nd and consequent builds
Call Stack (most recent call first):
  CMakeLists.txt:70 (include)


-- CMAKE_BUILD_TYPE: RelWithDebInfo
-- Performing Test HAS_RESERVED_IDENTIFIER
-- Performing Test HAS_RESERVED_IDENTIFIER - Success
-- Performing Test HAS_SUGGEST_DESTRUCTOR_OVERRIDE
-- Performing Test HAS_SUGGEST_DESTRUCTOR_OVERRIDE - Success
-- Performing Test HAS_SHADOW
-- Performing Test HAS_SHADOW - Success
-- Performing Test HAS_SUGGEST_OVERRIDE
-- Performing Test HAS_SUGGEST_OVERRIDE - Success
-- Performing Test HAS_USE_CTOR_HOMING
-- Performing Test HAS_USE_CTOR_HOMING - Success
-- Performing Test HAVE_SSSE3
-- Performing Test HAVE_SSSE3 - Success
-- Performing Test HAVE_SSE41
-- Performing Test HAVE_SSE41 - Success
-- Performing Test HAVE_SSE42
-- Performing Test HAVE_SSE42 - Success
-- Performing Test HAVE_PCLMULQDQ
-- Performing Test HAVE_PCLMULQDQ - Success
-- Performing Test HAVE_POPCNT
-- Performing Test HAVE_POPCNT - Success
-- Performing Test HAVE_AVX
-- Performing Test HAVE_AVX - Success
-- Performing Test HAVE_AVX2
-- Performing Test HAVE_AVX2 - Success
-- Performing Test HAVE_AVX512
-- Performing Test HAVE_AVX512 - Success
-- Performing Test HAVE_BMI
-- Performing Test HAVE_BMI - Success
-- Default libraries: -nodefaultlibs  -lc -lm -lpthread -ldl
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Unit tests are enabled
-- Building for: Darwin-21.4.0 x86_64  ;
    USE_STATIC_LIBRARIES=ON
    SPLIT_SHARED=
    CCACHE=CCACHE_FOUND-NOTFOUND 
-- Adding contrib module miniselect (configuring with miniselect-cmake)
-- Adding contrib module pdqsort (configuring with pdqsort-cmake)
-- Adding contrib module sparsehash-c11 (configuring with sparsehash-c11-cmake)
-- Adding contrib module abseil-cpp (configuring with abseil-cpp-cmake)
-- Adding contrib module magic_enum (configuring with magic-enum-cmake)
-- Adding contrib module boost (configuring with boost-cmake)
-- Adding contrib module cctz (configuring with cctz-cmake)
-- Packaging with tzdata version: 2021c
-- Adding contrib module consistent-hashing (configuring with consistent-hashing)
-- Adding contrib module dragonbox (configuring with dragonbox-cmake)
-- Adding contrib module hyperscan (configuring with hyperscan-cmake)
-- Adding contrib module jemalloc (configuring with jemalloc-cmake)
CMake Warning at contrib/jemalloc-cmake/CMakeLists.txt:20 (message):
  jemalloc support on non-linux is EXPERIMENTAL


-- jemalloc malloc_conf: oversize_threshold:0,muzzy_decay_ms:5000,dirty_decay_ms:5000
-- Adding contrib module libcpuid (configuring with libcpuid-cmake)
-- Adding contrib module libdivide (configuring with libdivide)
-- Adding contrib module libmetrohash (configuring with libmetrohash)
-- Adding contrib module lz4 (configuring with lz4-cmake)
-- Adding contrib module murmurhash (configuring with murmurhash)
-- Adding contrib module replxx (configuring with replxx-cmake)
-- Adding contrib module unixodbc (configuring with unixodbc-cmake)
-- ODBC is only supported on Linux
-- Not using ODBC
-- Adding contrib module nanodbc (configuring with nanodbc-cmake)
-- Adding contrib module capnproto (configuring with capnproto-cmake)
-- Adding contrib module yaml-cpp (configuring with yaml-cpp-cmake)
-- Adding contrib module re2 (configuring with re2-cmake)
-- Adding contrib module xz (configuring with xz-cmake)
-- Adding contrib module brotli (configuring with brotli-cmake)
-- Adding contrib module double-conversion (configuring with double-conversion-cmake)
-- Adding contrib module boringssl (configuring with boringssl-cmake)
-- Adding contrib module poco (configuring with poco-cmake)
-- Using Poco::Crypto
-- Not using Poco::Data::ODBC
-- Adding contrib module croaring (configuring with croaring-cmake)
-- Adding contrib module zstd (configuring with zstd-cmake)
-- ZSTD VERSION 1.5.0
-- Adding contrib module zlib-ng (configuring with zlib-ng-cmake)
-- Adding contrib module bzip2 (configuring with bzip2-cmake)
-- Adding contrib module minizip-ng (configuring with minizip-ng-cmake)
-- Adding contrib module snappy (configuring with snappy-cmake)
-- Adding contrib module rocksdb (configuring with rocksdb-cmake)
-- Performing Test HAVE_FALLOCATE
-- Performing Test HAVE_FALLOCATE - Failed
-- Performing Test HAVE_SYNC_FILE_RANGE_WRITE
-- Performing Test HAVE_SYNC_FILE_RANGE_WRITE - Failed
-- Performing Test HAVE_PTHREAD_MUTEX_ADAPTIVE_NP
-- Performing Test HAVE_PTHREAD_MUTEX_ADAPTIVE_NP - Failed
-- Looking for malloc_usable_size
-- Looking for malloc_usable_size - not found
-- Adding contrib module thrift (configuring with thrift-cmake)
-- Looking for arpa/inet.h
-- Looking for arpa/inet.h - found
-- Looking for fcntl.h
-- Looking for fcntl.h - found
-- Looking for getopt.h
-- Looking for getopt.h - found
-- Looking for inttypes.h
-- Looking for inttypes.h - found
-- Looking for netdb.h
-- Looking for netdb.h - found
-- Looking for netinet/in.h
-- Looking for netinet/in.h - found
-- Looking for signal.h
-- Looking for signal.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for sys/ioctl.h
-- Looking for sys/ioctl.h - found
-- Looking for sys/param.h
-- Looking for sys/param.h - found
-- Looking for sys/resource.h
-- Looking for sys/resource.h - found
-- Looking for sys/socket.h
-- Looking for sys/socket.h - found
-- Looking for sys/stat.h
-- Looking for sys/stat.h - found
-- Looking for sys/time.h
-- Looking for sys/time.h - found
-- Looking for sys/un.h
-- Looking for sys/un.h - found
-- Looking for poll.h
-- Looking for poll.h - found
-- Looking for sys/poll.h
-- Looking for sys/poll.h - found
-- Looking for sys/select.h
-- Looking for sys/select.h - found
-- Looking for sched.h
-- Looking for sched.h - found
-- Looking for string.h
-- Looking for string.h - found
-- Looking for strings.h
-- Looking for strings.h - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for gethostbyname_r
-- Looking for gethostbyname_r - not found
-- Looking for strerror_r
-- Looking for strerror_r - found
-- Looking for sched_get_priority_max
-- Looking for sched_get_priority_max - found
-- Looking for sched_get_priority_min
-- Looking for sched_get_priority_min - found
-- Performing Test STRERROR_R_CHAR_P
-- Performing Test STRERROR_R_CHAR_P - Failed
-- Adding contrib module arrow (configuring with arrow-cmake)
-- Looking for strtof_l
-- Looking for strtof_l - not found
-- Looking for strtoull_l
-- Looking for strtoull_l - not found
-- CMAKE_CXX_FLAGS:  -fdiagnostics-color=always -Xclang -fuse-ctor-homing -fsized-deallocation  -gdwarf-aranges -pipe -mssse3 -msse4.1 -msse4.2 -mpclmul -mpopcnt -fasynchronous-unwind-tables -ffile-prefix-map=/repos/haggy/ClickHouse=. -falign-functions=32   -Wall -Wno-unused-command-line-argument  -stdlib=libc++ -fdiagnostics-absolute-paths -fstrict-vtable-pointers -fexperimental-new-pass-manager -w -std=c++11 -Wall -pedantic -Werror -Wextra -Wno-unused-parameter -stdlib=libc++
-- Proceeding with version: 1.12.0.372
-- Adding contrib module avro (configuring with avro-cmake)
-- Adding contrib module protobuf (configuring with protobuf-cmake)
-- Adding contrib module openldap (configuring with openldap-cmake)
-- Adding contrib module grpc (configuring with grpc-cmake)
-- Performing Test IOS
-- Performing Test IOS - Failed
-- Performing Test IOS_V10
-- Performing Test IOS_V10 - Failed
-- Performing Test MACOS_V1012
-- Performing Test MACOS_V1012 - Success
-- Looking for clock_gettime in rt
-- Looking for clock_gettime in rt - not found
-- Looking for include file sys/types.h
-- Looking for include file sys/types.h - found
-- Looking for include file arpa/nameser_compat.h
-- Looking for include file arpa/nameser_compat.h - found
-- Looking for include file arpa/nameser.h
-- Looking for include file arpa/nameser.h - found
-- Looking for include file assert.h
-- Looking for include file assert.h - found
-- Looking for include file errno.h
-- Looking for include file errno.h - found
-- Looking for include file limits.h
-- Looking for include file limits.h - found
-- Looking for include file malloc.h
-- Looking for include file malloc.h - not found
-- Looking for include file memory.h
-- Looking for include file memory.h - found
-- Looking for include file netinet/tcp.h
-- Looking for include file netinet/tcp.h - found
-- Looking for include file net/if.h
-- Looking for include file net/if.h - found
-- Looking for include file socket.h
-- Looking for include file socket.h - not found
-- Looking for include file stdbool.h
-- Looking for include file stdbool.h - found
-- Looking for include file stdlib.h
-- Looking for include file stdlib.h - found
-- Looking for include file stropts.h
-- Looking for include file stropts.h - not found
-- Looking for include file sys/uio.h
-- Looking for include file sys/uio.h - found
-- Looking for include file time.h
-- Looking for include file time.h - found
-- Looking for include file dlfcn.h
-- Looking for include file dlfcn.h - found
-- Looking for include files winsock2.h, windows.h
-- Looking for include files winsock2.h, windows.h - not found
-- Looking for 3 include files winsock2.h, ..., windows.h
-- Looking for 3 include files winsock2.h, ..., windows.h - not found
-- Looking for include files winsock.h, windows.h
-- Looking for include files winsock.h, windows.h - not found
-- Looking for include file windows.h
-- Looking for include file windows.h - not found
-- Performing Test HAVE_SOCKLEN_T
-- Performing Test HAVE_SOCKLEN_T - Success
-- Performing Test HAVE_TYPE_SOCKET
-- Performing Test HAVE_TYPE_SOCKET - Failed
-- Performing Test HAVE_BOOL_T
-- Performing Test HAVE_BOOL_T - Success
-- Performing Test HAVE_SSIZE_T
-- Performing Test HAVE_SSIZE_T - Success
-- Performing Test HAVE_LONGLONG
-- Performing Test HAVE_LONGLONG - Success
-- Performing Test HAVE_SIG_ATOMIC_T
-- Performing Test HAVE_SIG_ATOMIC_T - Success
-- Performing Test HAVE_STRUCT_ADDRINFO
-- Performing Test HAVE_STRUCT_ADDRINFO - Success
-- Performing Test HAVE_STRUCT_IN6_ADDR
-- Performing Test HAVE_STRUCT_IN6_ADDR - Success
-- Performing Test HAVE_STRUCT_SOCKADDR_IN6
-- Performing Test HAVE_STRUCT_SOCKADDR_IN6 - Success
-- Performing Test HAVE_STRUCT_SOCKADDR_STORAGE
-- Performing Test HAVE_STRUCT_SOCKADDR_STORAGE - Success
-- Performing Test HAVE_STRUCT_TIMEVAL
-- Performing Test HAVE_STRUCT_TIMEVAL - Success
-- Looking for AF_INET6
-- Looking for AF_INET6 - found
-- Looking for O_NONBLOCK
-- Looking for O_NONBLOCK - found
-- Looking for FIONBIO
-- Looking for FIONBIO - found
-- Looking for SIOCGIFADDR
-- Looking for SIOCGIFADDR - found
-- Looking for MSG_NOSIGNAL
-- Looking for MSG_NOSIGNAL - found
-- Looking for PF_INET6
-- Looking for PF_INET6 - found
-- Looking for SO_NONBLOCK
-- Looking for SO_NONBLOCK - not found
-- Looking for CLOCK_MONOTONIC
-- Looking for CLOCK_MONOTONIC - found
-- Performing Test HAVE_SOCKADDR_IN6_SIN6_SCOPE_ID
-- Performing Test HAVE_SOCKADDR_IN6_SIN6_SCOPE_ID - Success
-- Performing Test HAVE_LL
-- Performing Test HAVE_LL - Success
-- Looking for bitncmp
-- Looking for bitncmp - not found
-- Looking for closesocket
-- Looking for closesocket - not found
-- Looking for CloseSocket
-- Looking for CloseSocket - not found
-- Looking for connect
-- Looking for connect - found
-- Looking for fcntl
-- Looking for fcntl - found
-- Looking for freeaddrinfo
-- Looking for freeaddrinfo - found
-- Looking for getaddrinfo
-- Looking for getaddrinfo - found
-- Looking for getenv
-- Looking for getenv - found
-- Looking for gethostbyaddr
-- Looking for gethostbyaddr - found
-- Looking for gethostname
-- Looking for gethostname - found
-- Looking for getnameinfo
-- Looking for getnameinfo - found
-- Looking for getservbyport_r
-- Looking for getservbyport_r - not found
-- Looking for gettimeofday
-- Looking for gettimeofday - found
-- Looking for if_indextoname
-- Looking for if_indextoname - found
-- Looking for inet_net_pton
-- Looking for inet_net_pton - found
-- Looking for inet_ntop
-- Looking for inet_ntop - found
-- Looking for inet_pton
-- Looking for inet_pton - found
-- Looking for ioctl
-- Looking for ioctl - found
-- Looking for ioctlsocket
-- Looking for ioctlsocket - not found
-- Looking for IoctlSocket
-- Looking for IoctlSocket - not found
-- Looking for recv
-- Looking for recv - found
-- Looking for recvfrom
-- Looking for recvfrom - found
-- Looking for send
-- Looking for send - found
-- Looking for setsockopt
-- Looking for setsockopt - found
-- Looking for socket
-- Looking for socket - found
-- Looking for strcasecmp
-- Looking for strcasecmp - found
-- Looking for strcmpi
-- Looking for strcmpi - not found
-- Looking for strdup
-- Looking for strdup - found
-- Looking for stricmp
-- Looking for stricmp - not found
-- Looking for strncasecmp
-- Looking for strncasecmp - found
-- Looking for strncmpi
-- Looking for strncmpi - not found
-- Looking for strnicmp
-- Looking for strnicmp - not found
-- Looking for writev
-- Looking for writev - found
-- Looking for __system_property_get
-- Looking for __system_property_get - not found
-- Adding contrib module msgpack-c (configuring with msgpack-c-cmake)
-- Adding contrib module cityhash102 (configuring with cityhash102)
-- Adding contrib module libfarmhash (configuring with libfarmhash)
-- Adding contrib module icu (configuring with icu-cmake)
-- Not using icu
-- Adding contrib module h3 (configuring with h3-cmake)
-- Adding contrib module mariadb-connector-c (configuring with mariadb-connector-c-cmake)
-- Build without mysqlclient (support for MYSQL dictionary source will be disabled)
-- Adding contrib module googletest (configuring with googletest-cmake)
-- Adding contrib module llvm (configuring with llvm-cmake)
-- Not using LLVM
-- Adding contrib module libxml2 (configuring with libxml2-cmake)
-- Adding contrib module aws;aws-c-common;aws-c-event-stream;aws-checksums (configuring with aws-s3-cmake)
-- Adding contrib module base64 (configuring with base64-cmake)
-- Adding contrib module simdjson (configuring with simdjson-cmake)
-- Adding contrib module rapidjson (configuring with rapidjson-cmake)
-- Adding contrib module fastops (configuring with fastops-cmake)
-- Not using fast vectorized mathematical functions library by Mikhail Parakhin
-- Adding contrib module libuv (configuring with libuv-cmake)
-- Adding contrib module AMQP-CPP (configuring with amqpcpp-cmake)
-- Adding contrib module cassandra (configuring with cassandra-cmake)
-- Adding contrib module fmtlib (configuring with fmtlib-cmake)
-- Adding contrib module krb5 (configuring with krb5-cmake)
-- Adding contrib module cyrus-sasl (configuring with cyrus-sasl-cmake)
-- Adding contrib module libgsasl (configuring with libgsasl-cmake)
-- Adding contrib module librdkafka (configuring with librdkafka-cmake)
-- librdkafka with SASL support
-- librdkafka with SSL support
-- Adding contrib module libhdfs3 (configuring with libhdfs3-cmake)
-- Not using hdfs
-- Adding contrib module hive-metastore (configuring with hive-metastore-cmake)
Hive disabled
-- Adding contrib module cppkafka (configuring with cppkafka-cmake)
-- Adding contrib module libpqxx (configuring with libpqxx-cmake)
-- Adding contrib module libpq (configuring with libpq-cmake)
-- Adding contrib module NuRaft (configuring with nuraft-cmake)
-- Adding contrib module fast_float (configuring with fast_float-cmake)
-- Adding contrib module datasketches-cpp (configuring with datasketches-cpp-cmake)
-- Adding contrib module libstemmer_c (configuring with libstemmer-c-cmake)
-- Adding contrib module wordnet-blast (configuring with wordnet-blast-cmake)
-- Adding contrib module lemmagen-c (configuring with lemmagen-c-cmake)
-- Adding contrib module nlp-data (configuring with nlp-data-cmake)
-- Adding contrib module cld2 (configuring with cld2-cmake)
-- Adding contrib module sqlite-amalgamation (configuring with sqlite-cmake)
-- Adding contrib module s2geometry (configuring with s2geometry-cmake)
-- Performing Test SUPPORTS_CXXFLAG_frame_larger_than=65536
-- Performing Test SUPPORTS_CXXFLAG_frame_larger_than=65536 - Success
-- Performing Test SUPPORTS_CFLAG_frame_larger_than=65536
-- Performing Test SUPPORTS_CFLAG_frame_larger_than=65536 - Success
-- Performing Test SUPPORTS_CXXFLAG_pedantic
-- Performing Test SUPPORTS_CXXFLAG_pedantic - Success
-- Performing Test SUPPORTS_CFLAG_pedantic
-- Performing Test SUPPORTS_CFLAG_pedantic - Success
-- Performing Test SUPPORTS_CXXFLAG_no_vla_extension
-- Performing Test SUPPORTS_CXXFLAG_no_vla_extension - Success
-- Performing Test SUPPORTS_CFLAG_no_vla_extension
-- Performing Test SUPPORTS_CFLAG_no_vla_extension - Success
-- Performing Test SUPPORTS_CXXFLAG_no_zero_length_array
-- Performing Test SUPPORTS_CXXFLAG_no_zero_length_array - Success
-- Performing Test SUPPORTS_CFLAG_no_zero_length_array
-- Performing Test SUPPORTS_CFLAG_no_zero_length_array - Success
-- Performing Test SUPPORTS_CXXFLAG_no_c11_extensions
-- Performing Test SUPPORTS_CXXFLAG_no_c11_extensions - Success
-- Performing Test SUPPORTS_CFLAG_no_c11_extensions
-- Performing Test SUPPORTS_CFLAG_no_c11_extensions - Success
-- Performing Test SUPPORTS_CXXFLAG_everything
-- Performing Test SUPPORTS_CXXFLAG_everything - Success
-- Performing Test SUPPORTS_CFLAG_everything
-- Performing Test SUPPORTS_CFLAG_everything - Success
-- Performing Test SUPPORTS_CXXFLAG_no_cxx98_compat_pedantic
-- Performing Test SUPPORTS_CXXFLAG_no_cxx98_compat_pedantic - Success
-- Performing Test SUPPORTS_CFLAG_no_cxx98_compat_pedantic
-- Performing Test SUPPORTS_CFLAG_no_cxx98_compat_pedantic - Success
-- Performing Test SUPPORTS_CXXFLAG_no_cxx98_compat
-- Performing Test SUPPORTS_CXXFLAG_no_cxx98_compat - Success
-- Performing Test SUPPORTS_CFLAG_no_cxx98_compat
-- Performing Test SUPPORTS_CFLAG_no_cxx98_compat - Success
-- Performing Test SUPPORTS_CXXFLAG_no_c99_extensions
-- Performing Test SUPPORTS_CXXFLAG_no_c99_extensions - Success
-- Performing Test SUPPORTS_CFLAG_no_c99_extensions
-- Performing Test SUPPORTS_CFLAG_no_c99_extensions - Success
-- Performing Test SUPPORTS_CXXFLAG_no_conversion
-- Performing Test SUPPORTS_CXXFLAG_no_conversion - Success
-- Performing Test SUPPORTS_CFLAG_no_conversion
-- Performing Test SUPPORTS_CFLAG_no_conversion - Success
-- Performing Test SUPPORTS_CXXFLAG_no_ctad_maybe_unsupported
-- Performing Test SUPPORTS_CXXFLAG_no_ctad_maybe_unsupported - Success
-- Performing Test SUPPORTS_CFLAG_no_ctad_maybe_unsupported
-- Performing Test SUPPORTS_CFLAG_no_ctad_maybe_unsupported - Success
-- Performing Test SUPPORTS_CXXFLAG_no_deprecated_dynamic_exception_spec
-- Performing Test SUPPORTS_CXXFLAG_no_deprecated_dynamic_exception_spec - Success
-- Performing Test SUPPORTS_CFLAG_no_deprecated_dynamic_exception_spec
-- Performing Test SUPPORTS_CFLAG_no_deprecated_dynamic_exception_spec - Success
-- Performing Test SUPPORTS_CXXFLAG_no_disabled_macro_expansion
-- Performing Test SUPPORTS_CXXFLAG_no_disabled_macro_expansion - Success
-- Performing Test SUPPORTS_CFLAG_no_disabled_macro_expansion
-- Performing Test SUPPORTS_CFLAG_no_disabled_macro_expansion - Success
-- Performing Test SUPPORTS_CXXFLAG_no_documentation_unknown_command
-- Performing Test SUPPORTS_CXXFLAG_no_documentation_unknown_command - Success
-- Performing Test SUPPORTS_CFLAG_no_documentation_unknown_command
-- Performing Test SUPPORTS_CFLAG_no_documentation_unknown_command - Success
-- Performing Test SUPPORTS_CXXFLAG_no_double_promotion
-- Performing Test SUPPORTS_CXXFLAG_no_double_promotion - Success
-- Performing Test SUPPORTS_CFLAG_no_double_promotion
-- Performing Test SUPPORTS_CFLAG_no_double_promotion - Success
-- Performing Test SUPPORTS_CXXFLAG_no_exit_time_destructors
-- Performing Test SUPPORTS_CXXFLAG_no_exit_time_destructors - Success
-- Performing Test SUPPORTS_CFLAG_no_exit_time_destructors
-- Performing Test SUPPORTS_CFLAG_no_exit_time_destructors - Success
-- Performing Test SUPPORTS_CXXFLAG_no_float_equal
-- Performing Test SUPPORTS_CXXFLAG_no_float_equal - Success
-- Performing Test SUPPORTS_CFLAG_no_float_equal
-- Performing Test SUPPORTS_CFLAG_no_float_equal - Success
-- Performing Test SUPPORTS_CXXFLAG_no_global_constructors
-- Performing Test SUPPORTS_CXXFLAG_no_global_constructors - Success
-- Performing Test SUPPORTS_CFLAG_no_global_constructors
-- Performing Test SUPPORTS_CFLAG_no_global_constructors - Success
-- Performing Test SUPPORTS_CXXFLAG_no_missing_prototypes
-- Performing Test SUPPORTS_CXXFLAG_no_missing_prototypes - Success
-- Performing Test SUPPORTS_CFLAG_no_missing_prototypes
-- Performing Test SUPPORTS_CFLAG_no_missing_prototypes - Success
-- Performing Test SUPPORTS_CXXFLAG_no_missing_variable_declarations
-- Performing Test SUPPORTS_CXXFLAG_no_missing_variable_declarations - Success
-- Performing Test SUPPORTS_CFLAG_no_missing_variable_declarations
-- Performing Test SUPPORTS_CFLAG_no_missing_variable_declarations - Success
-- Performing Test SUPPORTS_CXXFLAG_no_nested_anon_types
-- Performing Test SUPPORTS_CXXFLAG_no_nested_anon_types - Success
-- Performing Test SUPPORTS_CFLAG_no_nested_anon_types
-- Performing Test SUPPORTS_CFLAG_no_nested_anon_types - Success
-- Performing Test SUPPORTS_CXXFLAG_no_packed
-- Performing Test SUPPORTS_CXXFLAG_no_packed - Success
-- Performing Test SUPPORTS_CFLAG_no_packed
-- Performing Test SUPPORTS_CFLAG_no_packed - Success
-- Performing Test SUPPORTS_CXXFLAG_no_padded
-- Performing Test SUPPORTS_CXXFLAG_no_padded - Success
-- Performing Test SUPPORTS_CFLAG_no_padded
-- Performing Test SUPPORTS_CFLAG_no_padded - Success
-- Performing Test SUPPORTS_CXXFLAG_no_return_std_move_in_cxx11
-- Performing Test SUPPORTS_CXXFLAG_no_return_std_move_in_cxx11 - Failed
-- Performing Test SUPPORTS_CFLAG_no_return_std_move_in_cxx11
-- Performing Test SUPPORTS_CFLAG_no_return_std_move_in_cxx11 - Failed
-- Flag -Wno-return-std-move-in-c++11 is unsupported
-- Flag -Wno-return-std-move-in-c++11 is unsupported
-- Performing Test SUPPORTS_CXXFLAG_no_shift_sign_overflow
-- Performing Test SUPPORTS_CXXFLAG_no_shift_sign_overflow - Success
-- Performing Test SUPPORTS_CFLAG_no_shift_sign_overflow
-- Performing Test SUPPORTS_CFLAG_no_shift_sign_overflow - Success
-- Performing Test SUPPORTS_CXXFLAG_no_sign_conversion
-- Performing Test SUPPORTS_CXXFLAG_no_sign_conversion - Success
-- Performing Test SUPPORTS_CFLAG_no_sign_conversion
-- Performing Test SUPPORTS_CFLAG_no_sign_conversion - Success
-- Performing Test SUPPORTS_CXXFLAG_no_switch_enum
-- Performing Test SUPPORTS_CXXFLAG_no_switch_enum - Success
-- Performing Test SUPPORTS_CFLAG_no_switch_enum
-- Performing Test SUPPORTS_CFLAG_no_switch_enum - Success
-- Performing Test SUPPORTS_CXXFLAG_no_undefined_func_template
-- Performing Test SUPPORTS_CXXFLAG_no_undefined_func_template - Success
-- Performing Test SUPPORTS_CFLAG_no_undefined_func_template
-- Performing Test SUPPORTS_CFLAG_no_undefined_func_template - Success
-- Performing Test SUPPORTS_CXXFLAG_no_unused_template
-- Performing Test SUPPORTS_CXXFLAG_no_unused_template - Success
-- Performing Test SUPPORTS_CFLAG_no_unused_template
-- Performing Test SUPPORTS_CFLAG_no_unused_template - Success
-- Performing Test SUPPORTS_CXXFLAG_no_vla
-- Performing Test SUPPORTS_CXXFLAG_no_vla - Success
-- Performing Test SUPPORTS_CFLAG_no_vla
-- Performing Test SUPPORTS_CFLAG_no_vla - Success
-- Performing Test SUPPORTS_CXXFLAG_no_weak_template_vtables
-- Performing Test SUPPORTS_CXXFLAG_no_weak_template_vtables - Success
-- Performing Test SUPPORTS_CFLAG_no_weak_template_vtables
-- Performing Test SUPPORTS_CFLAG_no_weak_template_vtables - Success
-- Performing Test SUPPORTS_CXXFLAG_no_weak_vtables
-- Performing Test SUPPORTS_CXXFLAG_no_weak_vtables - Success
-- Performing Test SUPPORTS_CFLAG_no_weak_vtables
-- Performing Test SUPPORTS_CFLAG_no_weak_vtables - Success
-- compiler C   = /usr/local/opt/llvm/bin/clang -I/usr/local/opt/zlib/include -I/usr/local/opt/lbzip2/include -I/usr/local/opt/bzip2/include -I/usr/local/opt/sqlite/include -I/usr/local/opt/readline/include -I/usr/local/opt/openssl@1.1/include -I/usr/local/opt/snappy/include -I/usr/local/opt/libmemcached/include -I/usr/local/opt/mysql@5.7/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/include -fdiagnostics-color=always -Xclang -fuse-ctor-homing  -gdwarf-aranges -pipe -mssse3 -msse4.1 -msse4.2 -mpclmul -mpopcnt -fasynchronous-unwind-tables -ffile-prefix-map=./repos/haggy/ClickHouse=. -falign-functions=32  -Wall -Wno-unused-command-line-argument  -fdiagnostics-absolute-paths -fexperimental-new-pass-manager -Wframe-larger-than=65536 -Wpedantic -Wno-vla-extension -Wno-zero-length-array -Wno-c11-extensions -Weverything -Wno-c++98-compat-pedantic -Wno-c++98-compat -Wno-c99-extensions -Wno-conversion -Wno-ctad-maybe-unsupported -Wno-deprecated-dynamic-exception-spec -Wno-disabled-macro-expansion -Wno-documentation-unknown-command -Wno-double-promotion -Wno-exit-time-destructors -Wno-float-equal -Wno-global-constructors -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-nested-anon-types -Wno-packed -Wno-padded -Wno-shift-sign-overflow -Wno-sign-conversion -Wno-switch-enum -Wno-undefined-func-template -Wno-unused-template -Wno-vla -Wno-weak-template-vtables -Wno-weak-vtables -O2 -g -DNDEBUG -O3 -g -gdwarf-4 
-- compiler CXX = /usr/local/opt/llvm/bin/clang++  -fdiagnostics-color=always -Xclang -fuse-ctor-homing -fsized-deallocation  -gdwarf-aranges -pipe -mssse3 -msse4.1 -msse4.2 -mpclmul -mpopcnt -fasynchronous-unwind-tables -ffile-prefix-map=./repos/haggy/ClickHouse=. -falign-functions=32   -Wall -Wno-unused-command-line-argument  -stdlib=libc++ -fdiagnostics-absolute-paths -fstrict-vtable-pointers -fexperimental-new-pass-manager -Wextra -Wframe-larger-than=65536 -Wpedantic -Wno-vla-extension -Wno-zero-length-array -Wno-c11-extensions -Weverything -Wno-c++98-compat-pedantic -Wno-c++98-compat -Wno-c99-extensions -Wno-conversion -Wno-ctad-maybe-unsupported -Wno-deprecated-dynamic-exception-spec -Wno-disabled-macro-expansion -Wno-documentation-unknown-command -Wno-double-promotion -Wno-exit-time-destructors -Wno-float-equal -Wno-global-constructors -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-nested-anon-types -Wno-packed -Wno-padded -Wno-shift-sign-overflow -Wno-sign-conversion -Wno-switch-enum -Wno-undefined-func-template -Wno-unused-template -Wno-vla -Wno-weak-template-vtables -Wno-weak-vtables -O2 -g -DNDEBUG -O3 -g -gdwarf-4 
-- LINKER_FLAGS = -L/usr/local/opt/zlib/lib -L/usr/local/opt/lbzip2/lib -L/usr/local/opt/bzip2/lib -L/usr/local/opt/sqlite/lib -L/usr/local/opt/readline/lib -L/usr/local/opt/openssl@1.1/lib -L/usr/local/opt/snappy/lib -L/usr/local/opt/libmemcached/lib -L/usr/local/opt/mysql@5.7/lib -L/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/lib -rdynamic -Wl,-U,_inside_main 
-- ./repos/haggy/ClickHouse/src: Have 40278 megabytes of memory.
        Limiting concurrent linkers jobs to 11 and compiler jobs to 16 (system has 16 logical cores)
-- Will build ClickHouse 22.5.1.1 revision 54462 
INFONot generating debugger info for ClickHouse functions
-- StorageFileLog is only supported on Linux
-- ClickHouse modes:
-- Server mode: ON
-- Client mode: ON
-- Local mode: ON
-- Benchmark mode: ON
-- Extract from config mode: ON
-- Compressor mode: ON
-- Copier mode: ON
-- Format mode: ON
-- Obfuscator mode: ON
-- ODBC bridge mode: OFF
-- Library bridge mode: ON
-- ClickHouse install: ON
-- ClickHouse git-import: ON
-- ClickHouse keeper mode: ON
-- ClickHouse keeper-converter mode: ON
-- bash_completion will be written to /usr/local/share/bash-completion/completions
-- Target check already exists
-- /repos/haggy/ClickHouse/utils: Have 40278 megabytes of memory.
        Limiting concurrent linkers jobs to 11 and compiler jobs to OFF (system has 16 logical cores)
-- Configuring done
-- Generating done
-- Build files have been written to: ./repos/haggy/ClickHouse/build
```

### Build

```
#>> cmake --build . --config RelWithDebInfo

[704/7780] Building C object contrib/libpq-cmake/CMakeFiles/_libpq.dir/__/libpq/fe-secure-openssl.c.o
FAILED: contrib/libpq-cmake/CMakeFiles/_libpq.dir/__/libpq/fe-secure-openssl.c.o 
/usr/local/opt/llvm/bin/clang -DHAS_RESERVED_IDENTIFIER -isystem /Users/repos/haggy/ClickHouse/contrib/libpq -isystem /Users/repos/haggy/ClickHouse/contrib/libpq/include -isystem /Users/repos/haggy/ClickHouse/contrib/libpq/configs -isystem /Users/repos/haggy/ClickHouse/contrib/libcxx/include -isystem /Users/repos/haggy/ClickHouse/contrib/libcxx/src -isystem /Users/repos/haggy/ClickHouse/contrib/libcxxabi/include -isystem /Users/repos/haggy/ClickHouse/contrib/boringssl/include -I/usr/local/opt/zlib/include -I/usr/local/opt/lbzip2/include -I/usr/local/opt/bzip2/include -I/usr/local/opt/sqlite/include -I/usr/local/opt/readline/include -I/usr/local/opt/openssl@1.1/include -I/usr/local/opt/snappy/include -I/usr/local/opt/libmemcached/include -I/usr/local/opt/mysql@5.7/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/include -fdiagnostics-color=always -Xclang -fuse-ctor-homing  -gdwarf-aranges -pipe -mssse3 -msse4.1 -msse4.2 -mpclmul -mpopcnt -fasynchronous-unwind-tables -ffile-prefix-map=/Users/repos/haggy/ClickHouse=. -falign-functions=32  -Wall -Wno-unused-command-line-argument  -fdiagnostics-absolute-paths -fexperimental-new-pass-manager -w -O2 -g -DNDEBUG -O3 -g -gdwarf-4  -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk -mmacosx-version-min=10.15   -D OS_DARWIN -Werror -std=gnu11 -MD -MT contrib/libpq-cmake/CMakeFiles/_libpq.dir/__/libpq/fe-secure-openssl.c.o -MF contrib/libpq-cmake/CMakeFiles/_libpq.dir/__/libpq/fe-secure-openssl.c.o.d -o contrib/libpq-cmake/CMakeFiles/_libpq.dir/__/libpq/fe-secure-openssl.c.o -c /Users/repos/haggy/ClickHouse/contrib/libpq/fe-secure-openssl.c
/Users/repos/haggy/ClickHouse/contrib/libpq/fe-secure-openssl.c:1732:27: error: invalid application of 'sizeof' to an incomplete type 'BIO_METHOD' (aka 'struct bio_method_st')
                my_bio_methods = malloc(sizeof(BIO_METHOD));
                                        ^     ~~~~~~~~~~~~
/usr/local/Cellar/openssl@1.1/1.1.1n/include/openssl/bio.h:250:16: note: forward declaration of 'struct bio_method_st'
typedef struct bio_method_st BIO_METHOD;
               ^
/Users/repos/haggy/ClickHouse/contrib/libpq/fe-secure-openssl.c:1735:32: error: invalid application of 'sizeof' to an incomplete type 'BIO_METHOD' (aka 'struct bio_method_st')
                memcpy(my_bio_methods, biom, sizeof(BIO_METHOD));
                                             ^     ~~~~~~~~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/include/secure/_string.h:63:33: note: expanded from macro 'memcpy'
                __builtin___memcpy_chk (dest, __VA_ARGS__, __darwin_obsz0 (dest))
                                              ^~~~~~~~~~~
/usr/local/Cellar/openssl@1.1/1.1.1n/include/openssl/bio.h:250:16: note: forward declaration of 'struct bio_method_st'
typedef struct bio_method_st BIO_METHOD;
               ^
/Users/repos/haggy/ClickHouse/contrib/libpq/fe-secure-openssl.c:1736:17: error: incomplete definition of type 'struct bio_method_st'
                my_bio_methods->bread = my_sock_read;
                ~~~~~~~~~~~~~~^
/usr/local/Cellar/openssl@1.1/1.1.1n/include/openssl/bio.h:250:16: note: forward declaration of 'struct bio_method_st'
typedef struct bio_method_st BIO_METHOD;
               ^
/Users/repos/haggy/ClickHouse/contrib/libpq/fe-secure-openssl.c:1737:17: error: incomplete definition of type 'struct bio_method_st'
                my_bio_methods->bwrite = my_sock_write;
                ~~~~~~~~~~~~~~^
/usr/local/Cellar/openssl@1.1/1.1.1n/include/openssl/bio.h:250:16: note: forward declaration of 'struct bio_method_st'
typedef struct bio_method_st BIO_METHOD;
               ^
4 errors generated.
[706/7780] Building C object contrib/libpq-cmake/CMakeFiles/_libpq.dir/__/libpq/common/hmac_openssl.c.o
FAILED: contrib/libpq-cmake/CMakeFiles/_libpq.dir/__/libpq/common/hmac_openssl.c.o 
/usr/local/opt/llvm/bin/clang -DHAS_RESERVED_IDENTIFIER -isystem /Users/repos/haggy/ClickHouse/contrib/libpq -isystem /Users/repos/haggy/ClickHouse/contrib/libpq/include -isystem /Users/repos/haggy/ClickHouse/contrib/libpq/configs -isystem /Users/repos/haggy/ClickHouse/contrib/libcxx/include -isystem /Users/repos/haggy/ClickHouse/contrib/libcxx/src -isystem /Users/repos/haggy/ClickHouse/contrib/libcxxabi/include -isystem /Users/repos/haggy/ClickHouse/contrib/boringssl/include -I/usr/local/opt/zlib/include -I/usr/local/opt/lbzip2/include -I/usr/local/opt/bzip2/include -I/usr/local/opt/sqlite/include -I/usr/local/opt/readline/include -I/usr/local/opt/openssl@1.1/include -I/usr/local/opt/snappy/include -I/usr/local/opt/libmemcached/include -I/usr/local/opt/mysql@5.7/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/include -fdiagnostics-color=always -Xclang -fuse-ctor-homing  -gdwarf-aranges -pipe -mssse3 -msse4.1 -msse4.2 -mpclmul -mpopcnt -fasynchronous-unwind-tables -ffile-prefix-map=/Users/repos/haggy/ClickHouse=. -falign-functions=32  -Wall -Wno-unused-command-line-argument  -fdiagnostics-absolute-paths -fexperimental-new-pass-manager -w -O2 -g -DNDEBUG -O3 -g -gdwarf-4  -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk -mmacosx-version-min=10.15   -D OS_DARWIN -Werror -std=gnu11 -MD -MT contrib/libpq-cmake/CMakeFiles/_libpq.dir/__/libpq/common/hmac_openssl.c.o -MF contrib/libpq-cmake/CMakeFiles/_libpq.dir/__/libpq/common/hmac_openssl.c.o.d -o contrib/libpq-cmake/CMakeFiles/_libpq.dir/__/libpq/common/hmac_openssl.c.o -c /Users/repos/haggy/ClickHouse/contrib/libpq/common/hmac_openssl.c
/Users/repos/haggy/ClickHouse/contrib/libpq/common/hmac_openssl.c:93:23: error: invalid application of 'sizeof' to an incomplete type 'HMAC_CTX' (aka 'struct hmac_ctx_st')
        ctx->hmacctx = ALLOC(sizeof(HMAC_CTX));
                             ^     ~~~~~~~~~~
/Users/repos/haggy/ClickHouse/contrib/libpq/common/hmac_openssl.c:49:28: note: expanded from macro 'ALLOC'
#define ALLOC(size) malloc(size)
                           ^~~~
/usr/local/Cellar/openssl@1.1/1.1.1n/include/openssl/ossl_typ.h:102:16: note: forward declaration of 'struct hmac_ctx_st'
typedef struct hmac_ctx_st HMAC_CTX;
               ^
/Users/repos/haggy/ClickHouse/contrib/libpq/common/hmac_openssl.c:114:26: error: invalid application of 'sizeof' to an incomplete type 'HMAC_CTX' (aka 'struct hmac_ctx_st')
        memset(ctx->hmacctx, 0, sizeof(HMAC_CTX));
                                ^     ~~~~~~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/include/secure/_string.h:77:33: note: expanded from macro 'memset'
                __builtin___memset_chk (dest, __VA_ARGS__, __darwin_obsz0 (dest))
                                              ^~~~~~~~~~~
/usr/local/Cellar/openssl@1.1/1.1.1n/include/openssl/ossl_typ.h:102:16: note: forward declaration of 'struct hmac_ctx_st'
typedef struct hmac_ctx_st HMAC_CTX;
               ^
/Users/repos/haggy/ClickHouse/contrib/libpq/common/hmac_openssl.c:250:31: error: invalid application of 'sizeof' to an incomplete type 'HMAC_CTX' (aka 'struct hmac_ctx_st')
        explicit_bzero(ctx->hmacctx, sizeof(HMAC_CTX));
                                     ^     ~~~~~~~~~~
/usr/local/Cellar/openssl@1.1/1.1.1n/include/openssl/ossl_typ.h:102:16: note: forward declaration of 'struct hmac_ctx_st'
typedef struct hmac_ctx_st HMAC_CTX;
               ^
3 errors generated.
[721/7780] Building CXX object contrib/nuraft-cmake/CMakeFiles/_nuraft.dir/__/NuRaft/src/asio_service.cxx.o
ninja: build stopped: subcommand failed.
```

--- a/PreLoad.cmake
+++ b/PreLoad.cmake
@@ -19,6 +19,31 @@ if (NOT DEFINED ENV{CLION_IDE} AND NOT DEFINED ENV{XCODE_IDE})
     endif ()
 endif()
 
+# Check if environment is polluted.
+if (DEFINED ENV{CFLAGS} OR DEFINED ENV{CXXFLAGS} OR DEFINED ENV{LDFLAGS}
+    OR CMAKE_C_FLAGS OR CMAKE_CXX_FLAGS OR CMAKE_EXE_LINKER_FLAGS OR CMAKE_SHARED_LINKER_FLAGS OR CMAKE_MODULE_LINKER_FLAGS
+    OR CMAKE_C_FLAGS_INIT OR CMAKE_CXX_FLAGS_INIT OR CMAKE_EXE_LINKER_FLAGS_INIT OR CMAKE_SHARED_LINKER_FLAGS_INIT OR CMAKE_MODULE_LINKER_FLAGS_INIT)
+
+    message("CFLAGS: $ENV{CFLAGS}")
+    message("CXXFLAGS: $ENV{CXXFLAGS}")
+    message("LDFLAGS: $ENV{LDFLAGS}")
+    message("CMAKE_C_FLAGS: ${CMAKE_C_FLAGS}")
+    message("CMAKE_CXX_FLAGS: ${CMAKE_CXX_FLAGS}")
+    message("CMAKE_EXE_LINKER_FLAGS: ${CMAKE_EXE_LINKER_FLAGS}")
+    message("CMAKE_SHARED_LINKER_FLAGS: ${CMAKE_SHARED_LINKER_FLAGS}")
+    message("CMAKE_MODULE_LINKER_FLAGS: ${CMAKE_MODULE_LINKER_FLAGS}")
+
+    message(FATAL_ERROR "
+        Some of the variables like CFLAGS, CXXFLAGS, LDFLAGS are not empty.
+        It is not possible to build ClickHouse with custom flags.
+        These variables can be set up by previous invocation of some other build tools.
+        You should cleanup these variables and start over again.
+
+        Run the `env` command to check the details.
+        You will also need to remove the contents of the build directory.
+
+        Note: if you don't like this behavior, you can manually edit the cmake files, but please don't complain to developers.")
+endif()
 
 # Default toolchain - this is needed to avoid dependency on OS files.
 execute_process(COMMAND uname -s OUTPUT_VARIABLE OS)

